src/tools/nbench/benchmark.cpp

```diff
@@ -26,12 +26,56 @@
 using namespace std;
 using namespace ngraph;
 
+/// Utility to dump single HostTensor to std::out.
+template <class TYPE>
+static void dump_tensor_elements(runtime::HostTensor& tensor)
+{
+    TYPE* data_ptr = tensor.get_data_ptr<TYPE>();
+    size_t numElements = tensor.get_element_count();
+    for (size_t i = 0; i < numElements; ++i)
+    {
+        cout << data_ptr[i] << " ";
+    }
+    cout << endl;
+}
+
+/// Utility to dump all the result tensors to std::out.
+static void dump_result_tensors(vector<shared_ptr<runtime::HostTensor>>& results)
+{
+    cout << "============================================================================\n";
+    cout << "---- Dumping result tensors \n";
+    cout << "============================================================================\n";
+
+    unsigned i = 0;
+    for (auto& result : results)
+    {
+        cout << "Result tensor #" << i << ": " << result->get_name() << endl;
+        auto element_type = result->get_element_type();
+        switch (element_type)
```

Comment:
You should put the pragma that requires all element types are handled in this enum.
boolean is missing
I would also look at supporting bfloat16 and f16. I think you should be able to dump those too.